dask/dataframe/tests/test_groupby.py

```diff
@@ -2148,6 +2148,123 @@ def test_groupby_shift_with_freq():
     assert_eq(df_result, ddf.groupby("b").shift(periods=-2, freq="D", meta=df_result))
 
 
+@pytest.mark.parametrize("npartitions", [1, 3])
+@pytest.mark.parametrize("method", ["average", "min", "max", "first", "dense"])
+@pytest.mark.parametrize("ascending", [True, False])
+@pytest.mark.parametrize("na_option", ["keep", "top", "bottom"])
+@pytest.mark.parametrize("pct", [False, True])
+@pytest.mark.parametrize("axis", [0, 1])
+def test_groupby_rank_dataframe(npartitions, method, ascending, na_option, pct, axis):
+    pdf = pd.DataFrame(
+        {
+            "a": [0, 0, 1, 1, 2, 2, 3, 3, 3],
+            "b": [4, 4, 6, 3, 2, 1, None, 0, 0],
+            "c": [0, 0, 0, 0, 0, 1, 1, 1, 1],
+        },
+    )
+    ddf = dd.from_pandas(pdf, npartitions=npartitions)
+
+    with pytest.warns(UserWarning):
```

Comment:
Can you add a `match=` to this `pytest.warns` so that we know that we are getting exactly the warning that we expect? That will make the test easier to maintain over time.

Reply:
Done. Tests now check the warning informs the user that meta was not specified by default.